From 8305fcd90cdfbd5da2a4f581d13c7f0978e99e8e Mon Sep 17 00:00:00 2001
From: Angelo Arifi <arifilabs@users.noreply.github.com>
Date: Mon, 28 Sep 2026 22:14:34 -0400
Subject: [PATCH] Revert "night/sx8-numerics: revert GDN q/k norm to
 ggml_l2_norm (5fdfa62829, qwen35 only) - throwaway numerics probe"

This reverts commit ff323b02fdb1047b9562f538af2f08eabc43df4b.
---
 src/models/qwen35.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/models/qwen35.cpp b/src/models/qwen35.cpp
index d82bfa6bd5..74e45ff117 100644
--- a/src/models/qwen35.cpp
+++ b/src/models/qwen35.cpp
@@ -432,8 +432,8 @@ ggml_tensor * llama_model_qwen35::graph::build_layer_attn_linear(
 
     const float eps_norm = hparams.f_norm_rms_eps;
 
-    q_conv = ggml_l2_norm(ctx0, q_conv, eps_norm);
-    k_conv = ggml_l2_norm(ctx0, k_conv, eps_norm);
+    q_conv = build_gdn_l2_norm(ctx0, q_conv, eps_norm);
+    k_conv = build_gdn_l2_norm(ctx0, k_conv, eps_norm);
 
     //q_conv = ggml_cont_4d(ctx0, q_conv, head_k_dim, num_k_heads, n_seq_tokens, n_seqs);
     //k_conv = ggml_cont_4d(ctx0, k_conv, head_k_dim, num_k_heads, n_seq_tokens, n_seqs);
